src/metabase/query_processor/parameters.clj

```diff
@@ -0,0 +1,252 @@
+(ns metabase.query-processor.parameters
+  (:require [clojure.core.match :refer [match]]
+            [clojure.string :as s]
+            [clj-time.core :as t]
+            [clj-time.format :as tf]
+            [medley.core :as m]
+            [metabase.driver :as driver])
+  (:import (org.joda.time DateTimeConstants DateTime)))
+
+
+(def ^:private ^:const relative-dates
+  #{"today"
+    "yesterday"
+    "past7days"
+    "past30days"
+    "thisweek"
+    "thismonth"
+    "thisyear"
+    "lastweek"
+    "lastmonth"
+    "lastyear"})
+
+(defn- start-of-quarter [quarter year]
+  (condp = quarter
+    "Q1" (t/first-day-of-the-month (.withMonthOfYear (t/date-time year) DateTimeConstants/JANUARY))
+    "Q2" (t/first-day-of-the-month (.withMonthOfYear (t/date-time year) DateTimeConstants/APRIL))
+    "Q3" (t/first-day-of-the-month (.withMonthOfYear (t/date-time year) DateTimeConstants/JULY))
+    "Q4" (t/first-day-of-the-month (.withMonthOfYear (t/date-time year) DateTimeConstants/OCTOBER))))
```

Comment:
This would be a lot more concise if you made the `condp` the last argument of `.withMonthOfYear`. e.g.

```clojure
(defn- start-of-quarter [quarter year]
  (t/first-day-of-the-month (.withMonthOfYear (t/date-time year) (condp = quarter
                                                                   "Q1" DateTimeConstants/JANUARY
                                                                   "Q2" DateTimeConstants/APRIL
                                                                   "Q3" DateTimeConstants/JULY
                                                                   "Q4" DateTimeConstants/OCTOBER))))
```